src/transports/sentry_transport_curl.c

```diff
@@ -110,16 +111,24 @@ header_callback(char *buffer, size_t size, size_t nitems, void *userdata)
     return bytes;
 }
 
-static bool
-for_each_request_callback(sentry_prepared_http_request_t *req,
-    const sentry_envelope_t *UNUSED(envelope), void *data)
+static void
+task_exec_func(void *data)
 {
     struct task_state *ts = data;
+
+    sentry_prepared_http_request_t *req
+        = sentry__prepare_http_request(ts->envelope, ts->transport_state->rl);
+    ts->envelope = NULL;
+    if (!req) {
+        return;
```

Comment:
Consider logging an error here (same for WinHTTP)?

Reply:
It’s not really an error, this means the whole envelope was rate limited…

Reply:
You are right, I've missed that the rate limiting call inside. This means there are two significantly different error modes (rate limit vs http request preparation failed). In that case one of two suggestions:
1. Split up `prepare_http_request` from `rate_limit`, call them separately and log accordingly.
2. Log inside of `prepare_http_request`, but there are quite a few places where you exit.